### PR TITLE
Bug fix for conflict variable name in layernorm cells.

### PR DESCRIPTION
**Pull request by qlzh727**
# Description
Bug fix for conflict variable name in layernorm cells.

Brief Description of the PR:
The issue was caused by missing name scopes when invoking the build() method of sublayers, which is a quite common mistake. Usually the build() method is designed for user to implement, but not explicitly call.

Fixes # (issue)
https://github.com/tensorflow/addons/issues/2283
## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [x] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [x] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: tensorflow_addons/rnn/layer_norm_lstm_cell.py [hunk 1]
```diff
@@ -154,9 +154,16 @@ def __init__(
 
     def build(self, input_shape):
         super().build(input_shape)
-        self.kernel_norm.build([input_shape[0], self.units * 4])
-        self.recurrent_norm.build([input_shape[0], self.units * 4])
-        self.state_norm.build([input_shape[0], self.units])
+
+        def maybe_build_sublayer(sublayer, build_shape):
+            if not sublayer.built:
+                with tf.keras.backend.name_scope(sublayer.name):
```
Comment (bhack) [inline tensorflow_addons/rnn/layer_norm_lstm_cell.py:160]: I've not checked the internals but what is the difference with `maybe_init_scope` at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/base_layer.py#L2707

Comment (qlzh727) [reply tensorflow_addons/rnn/layer_norm_lstm_cell.py:160]: Usually the init scope is not needed by the cell since the RNN layer that wrap the cell should already have the correct context before the cell.build() is invoked (eg by __call__() of the model, or the layer). Manually invoke the build() method is not ideal and often error-prone if the context is not setup properly.

**Review by seanpmorgan — APPROVED** [review]
LGTM Thanks!
